tests/test_target.py

```diff
@@ -14,35 +15,69 @@
 # limitations under the License.
 #
 
+""" Module for testing different targets. """
+
 import os
-import shutil
-import tempfile
 from unittest import TestCase
-
-from devlib import LocalLinuxTarget
-
+from devlib import AndroidTarget, LinuxTarget, LocalLinuxTarget
+from devlib.utils.misc import get_random_string
 
 class TestReadTreeValues(TestCase):
+    """ Class testing Target.read_tree_values_flat() """
+
+    def test_read_multiline_values(self, target):
+        """ Test Target.read_tree_values_flat() """
+        method_name = self.__class__.test_read_multiline_values.__qualname__
 
-    def test_read_multiline_values(self):
         data = {
             'test1': '1',
             'test2': '2\n\n',
             'test3': '3\n\n4\n\n',
         }
 
-        tempdir = tempfile.mkdtemp(prefix='devlib-test-')
+        dirname = os.path.join(target.working_directory,
+                               f'devlib-test-{get_random_string(12)}')
+        print(f'{method_name}: creating {dirname}...')
+        target.makedirs(dirname)
+
         for key, value in data.items():
-            path = os.path.join(tempdir, key)
-            with open(path, 'w') as wfh:
-                wfh.write(value)
+            path = os.path.join(dirname, key)
+            str_value = value.replace("\n", "\\n")
```

Comment:
that's fishy, why do we need that ? If you want to have the string displayed as a Python literal, you can use the `repr()` of the string, which can be done with `{value!r}` in the f-string.

Reply:
I just wanted to log what is written as `value`. `{value!r}` is really cool. Thanks.